lib/mix/lib/mix/tasks/test.ex

```diff
@@ -225,7 +225,7 @@ defmodule Mix.Tasks.Test do
     ex_unit_opts = ex_unit_opts(opts)
     ExUnit.configure(ex_unit_opts)
 
-    test_paths = project[:test_paths] || ["test"]
+    test_paths = project[:test_paths] || default_test_paths
```

Comment:
You need parentheses when calling `default_test_paths/1`, it emits a warning otherwise that you can see on Travis as well :)